File: app/src/main/java/activity/amigosecreto/adapter/ParticipantesRecyclerAdapter.java
package activity.amigosecreto.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.AnimationUtils;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import activity.amigosecreto.R;
import activity.amigosecreto.db.Participante;

/**
 * Modern RecyclerView Adapter for Participantes with animations
 */
public class ParticipantesRecyclerAdapter extends RecyclerView.Adapter<ParticipantesRecyclerAdapter.ViewHolder> {

    private Context context;
    private List<Participante> participantes;
    private OnItemClickListener listener;
    private int lastPosition = -1;

    public interface OnItemClickListener {
        void onItemClick(Participante participante);
        void onRemoveClick(Participante participante);
        void onShareClick(Participante participante);
    }

    public ParticipantesRecyclerAdapter(Context context, List<Participante> participantes) {
        this.context = context;
        this.participantes = participantes;
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_participante, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Participante participante = participantes.get(position);
        holder.bind(participante);

        // Apply animation
        if (position > lastPosition) {
            holder.itemView.startAnimation(AnimationUtils.loadAnimation(context, R.anim.card_appear));
            lastPosition = position;
        }
    }

    @Override
    public int getItemCount() {
        return participantes.size();
    }

    public void updateList(List<Participante> newList) {
        this.participantes = newList;
        notifyDataSetChanged();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvNome, tvEmail, tvNumero, tvAvatar;
        ImageButton btnRemover, btnShare;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNome = itemView.findViewById(R.id.tv_nome);
            tvEmail = itemView.findViewById(R.id.tv_email);
            tvNumero = itemView.findViewById(R.id.tv_numero);
            tvAvatar = itemView.findViewById(R.id.tv_avatar);
            btnRemover = itemView.findViewById(R.id.btn_remover);
            btnShare = itemView.findViewById(R.id.btn_share);
        }

        void bind(Participante participante) {
            tvNome.setText(participante.getNome());
            tvNumero.setText(String.valueOf(getAdapterPosition() + 1));

            // Avatar with first letter
            String inicial = participante.getNome().substring(0, 1).toUpperCase();
            tvAvatar.setText(inicial);

            // Status
            if (participante.isEnviado()) {
                tvEmail.setText("✓ Enviado");
            } else {
                tvEmail.setText("Pendente");
            }

            // Click listeners
            itemView.setOnClickListener(v -> {
                if (listener != null) {
                    listener.onItemClick(participante);
                }
            });

            btnRemover.setOnClickListener(v -> {
                if (listener != null) {
                    listener.onRemoveClick(participante);
                }
            });

            btnShare.setOnClickListener(v -> {
                if (listener != null) {
                    listener.onShareClick(participante);
                }
            });
        }
    }
}

File: app/src/main/java/activity/amigosecreto/adapter/ParticipantesRecyclerAdapter.kt
package activity.amigosecreto.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import activity.amigosecreto.R
import activity.amigosecreto.db.Participante

/**
 * Modern RecyclerView Adapter for Participantes with animations
 */
class ParticipantesRecyclerAdapter(
    private val context: Context,
    private var participantes: List<Participante>
) : RecyclerView.Adapter<ParticipantesRecyclerAdapter.ViewHolder>() {

    private var listener: OnItemClickListener? = null
    private var lastPosition = -1

    interface OnItemClickListener {
        fun onItemClick(participante: Participante)
        fun onRemoveClick(participante: Participante)
        fun onShareClick(participante: Participante)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_participante, parent, false)
        return ViewHolder(view)
    }

    @Suppress("RecyclerView")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val participante = participantes[position]
        holder.bind(participante)

        // Apply animation
        if (position > lastPosition) {
            holder.itemView.startAnimation(AnimationUtils.loadAnimation(context, R.anim.card_appear))
            lastPosition = position
        }
    }

    override fun getItemCount() = participantes.size

    @Suppress("NotifyDataSetChanged")
    fun updateList(newList: List<Participante>) {
        participantes = newList
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvNome: TextView = itemView.findViewById(R.id.tv_nome)
        val tvEmail: TextView = itemView.findViewById(R.id.tv_email)
        val tvNumero: TextView = itemView.findViewById(R.id.tv_numero)
        val tvAvatar: TextView = itemView.findViewById(R.id.tv_avatar)
        val btnRemover: ImageButton = itemView.findViewById(R.id.btn_remover)
        val btnShare: ImageButton = itemView.findViewById(R.id.btn_share)

        fun bind(participante: Participante) {
            tvNome.text = participante.nome
            tvNumero.text = (bindingAdapterPosition + 1).toString()

            // Avatar with first letter
            // TODO: extrair para fun String?.toAvatarText() — mesma expressão duplicada em
            //       ExclusionViewHolder e no ViewHolder inline do bottom sheet em ParticipantesActivity.
            tvAvatar.text = participante.nome?.trim()?.firstOrNull()?.uppercaseChar()?.toString() ?: "?"

            // Status
            tvEmail.text = if (participante.isEnviado) context.getString(R.string.status_item_sent) else context.getString(R.string.status_item_pending)

            // Click listeners
            itemView.setOnClickListener {
                listener?.onItemClick(participante)
            }

            btnRemover.setOnClickListener {
                listener?.onRemoveClick(participante)
            }

            btnShare.setOnClickListener {
                listener?.onShareClick(participante)
            }
        }
    }
}
